src/rdd/rdd.rs

```diff
@@ -568,6 +568,67 @@ pub trait Rdd: RddBase + 'static {
         Ok(buf)
     }
 
+    /// Randomly splits this RDD with the provided weights.
+    fn random_split(
+        &self,
+        weights: Vec<f64>,
+        _seed: Option<u64>,
+    ) -> Vec<SerArc<dyn Rdd<Item = Self::Item>>>
+    where
+        Self: Sized,
+    {
+        let sum: f64 = weights.iter().sum();
+        assert!(
+            weights.iter().all(|&x| x >= 0.0),
+            format!("Weights must be nonnegative, but got {:?}", weights)
+        );
+        assert!(
+            sum > 0.0,
+            format!("Sum of weights must be positive, but got {:?}", weights)
+        );
+
+        let seed: u64 = match _seed {
+            Some(x) => x,
+            None => rand::random::<u64>(),
+        };
+
+
+        let mut full_bounds = vec![0.0f64];
+        let bounds: Vec<f64> =
+            weights
+                .into_iter()
+                .map(|weight| weight / sum)
+                .collect::<Vec<f64>>()
+                .iter()
+                .scan(0.0f64, |state, &x| {
+                    *state = *state + x;
+                    Some(*state)
+                })
+                .collect();
+        full_bounds.extend(bounds);
+
+        let mut splitted_rdds: Vec<SerArc<dyn Rdd<Item = Self::Item>>> = Vec::new();
+
+        for bound in full_bounds.windows(2) {
```

Comment:
I think you were trying to replicate how is done with the partitionwise rdd. The difference here is that you can't hook directly on the compute method and tap on the `get_sampler` method since you are directly passing the serialized closure to a map partitions rdd. However I think something similar is possible, since in this closure you are receiving a given partition items iterator you could do something along:
```
Fn!(|partition: Box<dyn Iterator ...> {
let bcs = BernoulliCellSampler::new(lower_bound, upper_bound, false);
let sampler_func = bcs.get_sampler();
Box::new(sampler_func(partition))
})
```

this way we could keep a consistent API.

Reply:
Initially I did try to implement the RandomSampler#get_sampler() for BCS, but was struggling with borrow checker and the trait system. Again would love your help to refactor it.

Reply:
Just tried again with approach, however the RandomSampler::get_sampler() signature requires only the &self, however the BCS requires custom Pcg64 to generate sample. So to do this will break the existing Samplers features, which is something I am not quite confident to do so yet.

Reply:
Did the changes to reuse `get_sampler` as suggested. As to the `RamdomSampler` trait I am still trying to figuring out, but feel free to start with another commit about this. :)